[Swift in WebKit] The Swift implementation of WKRKEntity is incorrect

rr-codes opened the pull request:
#### dbc4806d5b416e9968a4b5464783f423d05741fc
<pre>
[Swift in WebKit] The Swift implementation of WKRKEntity is incorrect
<a href="https://bugs.webkit.org/show_bug.cgi?id=293885">https://bugs.webkit.org/show_bug.cgi?id=293885</a>
<a href="https://rdar.apple.com/152409392">rdar://152409392</a>

Reviewed by Ada Chan.

Use `@objc @implementation` to properly implement WKRKEntity in Swift, and address the issues it exposes.

Also rename the type for more consistency.

Subsequent commits will format/lint the code and address the concurrency issues.

* Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.h:
* Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.mm:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/ios/WKPageHostedModelView.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.h: Renamed from Source/WebKit/WebKitSwift/RealityKit/RealityKitBridging.h.
* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.swift: Renamed from Source/WebKit/WebKitSwift/RealityKit/RKEntity.swift.
* Source/WebKit/WebKitSwift/StageMode/WKStageMode.h:
* Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift:
(WKStageModeInteractionDriver.setContainerTransformInPortal):
(WKStageModeInteractionDriver.operationDidUpdate(_:)):
* Source/WebKit/WebKitSwift/WebKitSwift.h:

Canonical link: <a href="https://commits.webkit.org/295755@main">https://commits.webkit.org/295755@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8ca544b389ff5e8a49a9eb62cdb9cbcbdb7726da

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105992 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25741 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16135 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111188 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56588 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26391 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34244 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80513 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108996 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20819 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95661 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60833 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20428 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56026 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90199 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13794 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-pseudo/highlight-custom-properties-dynamic-001.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114042 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33130 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24445 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89593 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33494 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91889 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89274 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34141 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11961 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28675 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17200 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33055 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38467 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32801 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36150 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34399 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achan00 commented on Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.mm:
```diff
@@ -33,7 +33,7 @@
 #import "ModelProcessModelPlayerManagerProxy.h"
 #import "ModelProcessModelPlayerMessages.h"
-#import "RealityKitBridging.h"
+#import "WKRKEntity.h"
```
Nit: ordering

achan00 commented on Source/WebKit/WebKitSwift/WebKitSwift.h:
```diff
@@ -24,7 +24,7 @@
 
 #import "LinearMediaKitExtras.h"
-#import "RealityKitBridging.h"
+#import "WKRKEntity.h"
```
Nit: ordering

achan00 reviewed (approved):
Thanks for cleaning this up.

rr-codes committed:
[Swift in WebKit] The Swift implementation of WKRKEntity is incorrect
https://bugs.webkit.org/show_bug.cgi?id=293885
rdar://152409392

Reviewed by Ada Chan.

Use `@objc @implementation` to properly implement WKRKEntity in Swift, and address the issues it exposes.

Also rename the type for more consistency.

Subsequent commits will format/lint the code and address the concurrency issues.

* Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.h:
* Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.mm:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/ios/WKPageHostedModelView.mm:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.h: Renamed from Source/WebKit/WebKitSwift/RealityKit/RealityKitBridging.h.
* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.swift: Renamed from Source/WebKit/WebKitSwift/RealityKit/RKEntity.swift.
* Source/WebKit/WebKitSwift/StageMode/WKStageMode.h:
* Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift:
(WKStageModeInteractionDriver.setContainerTransformInPortal):
(WKStageModeInteractionDriver.operationDidUpdate(_:)):
* Source/WebKit/WebKitSwift/WebKitSwift.h:

Canonical link: https://commits.webkit.org/295755@main
